pkg/param/pagination/pagination.go

```diff
@@ -1,30 +1,46 @@
 package pagination
 
-type GetPage struct {
+// Page controls enable iterative retrieval of a large data set. This
+// structure is used both on input, to specify the boundaries of a
+// page to retrieve, and on output, to indicate the items actually
+// retrieved. The TotalRows value is ignored on input
+type Page struct {
 	StartRow  *int
 	EndRow    *int
-	PageSize  *int
 	TotalRows *int
 }
 
-func (page *GetPage) ContainsMore() bool {
-
-	if *page.EndRow < *page.TotalRows {
-		return true
-	}
-	return false
-}
-func (page *GetPage) SetStartRow(row int) {
+// SetStartRow - Specify the first row to retrieve in a page of results, using a zero-based index.
+func (page *Page) SetStartRow(row int) {
 	page.StartRow = &row
 }
 
-func (page *GetPage) SetEndRow(row int) {
+// SetEndRow - Specify the first row NOT to retrieve in a page of results, using a zero-based index
+// value one larger than the last row desired.
+func (page *Page) SetEndRow(row int) {
 	page.EndRow = &row
 }
-func (page *GetPage) SetTotalRows(row int) {
-	page.TotalRows = &row
+
+// SetPageSize - set the page size for batch processing
+func (page *Page) SetPageSize(psize int) {
+	zero := 0
+	page.StartRow = &zero
+	page.EndRow = &psize
 }
 
-func (page *GetPage) SetPageSize(row int) {
-	page.PageSize = &row
+// NextPage - Returns true if objects are more than pagesize.
```

Comment:
I suggest:

  // Updates the StartRow and EndRow values to request the next page of the
  // same size, if more rows are available according to the TotalRows value.
  // Returns true if the values are actually updated.